packages/web-components/fast-foundation/src/data-grid/data-grid.ts

```diff
@@ -184,6 +207,64 @@ export class FASTDataGrid extends FASTElement {
         }
     }
 
+    /**
+     * Defines how the grid handles row or cell selection.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: selection-mode
+     */
+    @attr({ attribute: "selection-mode" })
+    public selectionMode: DataGridSelectionMode = "none";
+    private selectionModeChanged(
+        prev: DataGridSelectionMode,
+        next: DataGridSelectionMode
+    ): void {
+        if (this.$fastController.isConnected) {
+            if (prev === "single-row" || prev === "multi-row") {
+                this.removeEventListener(
+                    "rowselectionchange",
+                    this.handleRowSelectedChange
+                );
+            }
+            if (next === "single-row" || next === "multi-row") {
+                this.addEventListener("rowselectionchange", this.handleRowSelectedChange);
+            }
+            this.deselectAllRows();
+        }
+    }
+
+    /**
+     * Determines if clicks can automatically select
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: disable-click-select
+     */
+    @attr({ attribute: "disable-click-select", mode: "boolean" })
+    public disableClickSelect: boolean = false;
```

Comment:
I don't love the name here. I get why a Boolean property needs to default to false, and this feature makes sense to enable by default. Additionally, it's not just a "click" but also keyboard interaction.

What about calling this "managed" selection, so at least "disable-managed-selection".

Or "selection-behavior" = "managed" | "auto" or "manual".

Reply:
It's actually not also the keyboard interaction.  How you select/deselect a row via keyboard is defined in the accessibility spec [here](https://w3c.github.io/aria-practices/#keyboard-interaction-for-data-grids).  This just disables our default "click anywhere in the row and we select it" behavior so that authors can link it to a checkbox in a column, for example.  This behavior is described in the spec:

"Shift + Space: Selects the row that contains the focus. If the grid includes a column with checkboxes for selecting rows, this key can serve as a shortcut for checking the box when focus is not on the checkbox."  
(note: fixing the issue where I'm not checking for Shift on select with space key)

So, while I do agree we should enable disabling keyboard selection as well, authors should be able to disable mouse selection only.  So, your "selection-behavior" would have to be something like - "programmatic only", "keyboard", "keyboard+mouse".

Reply:
Got it. I didn't connect all that but it makes sense. Something still feels funny in the name, but I'm not sure what it is or a better option. It might just be the form of the wording, like "disable click TO select" or "disable click selectION". I'm fine with this unless you like one of those better.

Reply:
So updated with a more flexible "selection-behavior" enum?